src/tests/ftest/container/boundary.py

```diff
@@ -0,0 +1,144 @@
+        self.log.info("==(2.%d) %d containers created.", pool_num, num_containers)
+
+        # Launch the create_container_and_test threads
+        self.log.info("==Launching %d create_container_and_test threads", thread_manager.qty)
+        failed_thread_count = thread_manager.check_run()
+        if failed_thread_count > 0:
+            msg = "#(2.{}) FAILED create_container_and_test Threads".format(failed_thread_count)
+            self.d_log.error(msg)
+            self.fail(msg)
+
+    def create_pools(self, num_pools=10, num_containers=100, with_io=False):
+        """To create number of pools and containers parallelly.
+        Args:
+            num_pools (int): number of pools to create.
+            num_containers (int): number of containers to create.
+            with_io (bool): enable container test with execute_io.
+        """
+
+        # Setup the thread manager
+        thread_manager = ThreadManager(self.create_containers, self.timeout - 30)
+
+        for pool_number in range(num_pools):
+            pool = self.get_pool()
+            thread_manager.add(
+                pool=pool, pool_num=pool_number, num_containers=num_containers, with_io=with_io)
+            self.log.info("=(1.%d) pool created, %d.", pool_number, pool)
+
+        # Launch the create_containers threads
+        self.log.info("=Launching %d create_containers threads", thread_manager.qty)
+        failed_thread_count = thread_manager.check_run()
+        if failed_thread_count > 0:
+            msg = "#(1.{}) FAILED create_containers Threads".format(failed_thread_count)
+            self.d_log.error(msg)
+            self.fail(msg)
+
+    def test_container_boundary(self):
+        """JIRA ID: DAOS-8464 Test lots of pools and containers in parallel.
+        Test Description:
+            Testcase 1: Test 1 pool with containers boundary condition in parallel.
+            Testcase 2: Test large number of pools and containers in parallel.
+            Testcase 3: Test pools and containers with io.
+            log.info: (a.b.c) a: test-step,  b: pool_number,  c: container_number
+        Use case:
+            0. Bring up DAOS server.
+            1. Create pools by ThreadManager.
+            2. Create containers under each pool by sub ThreadManager.
+            3. Launch io and syncup each container.
+            4. Close container.
+        :avocado: tags=all,full_regression
+        :avocado: tags=container, pool
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=boundary_test
+        """
+
+        num_pools = self.params.get("num_pools", '/run/boundary_test/*')
+        num_containers = self.params.get("num_containers", '/run/boundary_test/*')
+        io = self.params.get("with_io", '/run/boundary_test/*')
+        pool_cont_args = {
+            "num_pools": num_pools,
+            "num_containers": num_containers,
+            "with_io": io
+        }
+        self.create_pools(**pool_cont_args)
```

Comment:
Minor style - is there a specific reason to use a dict and `**pool_cont_args`? Why not just this?
```suggestion
        self.create_pools(
            num_pools=num_pools,
            num_containers=num_containers,
            with_io=io)
```

Reply:
Updated.